lib/datastore.js

```diff
@@ -598,16 +598,30 @@ class Datastore extends EventEmitter {
    */
   _getRawCandidates (query) {
     const indexNames = Object.keys(this.indexes)
+
+    const filterIndexNames = (indexNames) => ([k, v]) => !!(typeof v === 'string' || typeof v === 'number' || typeof v === 'boolean' || isDate(v) || v === null) &&
```

Comment:
Is there a reason it is defined within `_getRawCandidates`? It'd be (barely, I admit) more efficient if it was defined within the utils file, no?

Reply:
done